### PR TITLE
Proofread Oracle SQL - SET TRANSACTION

### PR DESCRIPTION
**Pull request by Liangda-w**
Relates #10513 .

Changes proposed in this pull request:
- Proofread Oracle SQL - [SET TRANSACTION](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/SET-TRANSACTION.html#GUID-F11E1E30-5871-48D1-8266-F80A1DF126A1)
- Add corresponding test cases

### CODE REVIEW

**Comment by tristaZero** [comment]
Hey, @wgy8283335, Please give some attention here.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/TCLStatement.g4 [hunk 1]
```diff
@@ -20,7 +20,10 @@ grammar TCLStatement;
 import Symbol, Keyword, OracleKeyword, Literals, BaseRule;
 
 setTransaction
-    : SET TRANSACTION
+    : SET TRANSACTION ((READ (ONLY | WRITE)
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/TCLStatement.g4:23]: The rule is not correct.
image

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml [hunk 2]
```diff
@@ -21,6 +21,10 @@
     <sql-case id="setGlobalTransaction" value="SET GLOBAL TRANSACTION ISOLATION LEVEL REPEATABLE READ" db-types="MySQL" />
     <sql-case id="setSessionTransaction" value="SET SESSION CHARACTERISTICS AS TRANSACTION ISOLATION LEVEL READ COMMITTED" db-types="PostgreSQL" />
     <sql-case id="setTransactionReadOnly" value="SET TRANSACTION READ ONLY" db-types="MySQL,Oracle" />
+    <sql-case id="setTransactionReadWriteWithName" value="SET TRANSACTION READ WRITE NAME 'Toronto'" db-types="Oracle" />
+    <sql-case id="setTransactionIsolationLevelSerializable" value="SET TRANSACTION ISOLATION LEVEL SERIALIZABLE" db-types="Oracle" />
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml:25]: Lack the keywords ‘NAME string’.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml [hunk 3]
```diff
@@ -21,6 +21,10 @@
...
+    <sql-case id="setTransactionIsolationLevelReadCommitted" value="SET TRANSACTION ISOLATION LEVEL READ COMMITTED" db-types="Oracle" />
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml:26]: Lack the keywords ‘NAME string’.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml [hunk 4]
```diff
@@ -21,6 +21,10 @@
...
+    <sql-case id="setTransactionUseRollbackSegment" value="SET TRANSACTION USE ROLLBACK SEGMENT rbs_ts" db-types="Oracle" />
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml:27]: Lack the keywords ‘NAME string’.

**Review by wgy8283335 — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/TCLStatement.g4 (hunk 1 above)
Comment (Liangda-w) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/TCLStatement.g4:23]: Hi @wgy8283335, could you please specify which point and why it's not correct?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml (hunk 2 above)
Comment (Liangda-w) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/tcl/set-transaction.xml:25]: `NAME string` is optional here, so I didn't add it.

**Review by wgy8283335 — APPROVED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/TCLStatement.g4 (hunk 1 above)
Comment (wgy8283335) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/TCLStatement.g4:23]: It's my miss understanding.

**Comment by tristaZero** [comment]
Hi @Liangda-w ,

Thanks for this PR. Plus, **please have my congratulations on your accepted GSoC 2021 project!** 🎉 

Since we received a boom of proposals I was a little worried your proposal could not pass, although I scored your proposal higher. Fortunately, you did it, congrats!

In the following months, @wgy8283335  and I will provide our mentoring as possible as we can to help you cross the final line of this project. Meanwhile, we hope you can keep in touch with this community and finish those tasks basing on your proposal schedule.

We kindly hope you can have a nice journey of GSoC 2021 and stay here even after it finishes. 👨‍👦‍👦 
If you have any questions, please be free to contact @wgy8283335  and me here or by email.

Best,
Trista

**Pull request merged**